django/http/multipartparser.py

```diff
@@ -754,3 +754,11 @@ def _parse_header_params(s):
         plist.append(f.strip())
         s = s[end:]
     return plist
+
+
+def valid_boundary(s):
+    if isinstance(s, bytes):
+        _vb_pattern = b"^[ -~]{0,200}[!-~]$"
+    else:
+        _vb_pattern = "^[ -~]{0,200}[!-~]$"
+    return re.match(_vb_pattern, s)
```

Comment:
`s` is always `bytes` in our case so we can drop the `else` branch:
```suggestion
    return re.match(b"^[ -~]{0,200}[!-~]$", s)
```
We could also pre-compile this regex and use `fullmatch()`:
```python
boundary_re = _lazy_re_compile(b"[ -~]{0,200}[!-~]")

def valid_boundary(s):
    return boundary_re.fullmatch(s)
```
It's used only in `MultiPartParser` so maybe a separate hook is not necessary :thinking: 
```python
if not boundary or not boundary_re.fullmatch(boundary):
    raise MultiPartParserError(
        "Invalid boundary in multipart: %s" % force_str(boundary)
    )
```

Reply:
👍 See fa85411701f7c5ef5802f377a1e5314144be48e5. 

* I made it a class attribute. (It's clearly then a multipart boundary re...) 
* I just went with `re.compile` — I don't see the benefit of deferring it? 🤔